sdks/python/apache_beam/coders/coder_impl.py

```diff
@@ -161,10 +161,10 @@ def get_estimated_size_and_observables(self, value, nested=False):
       return self.estimate_size(value, nested), []
 
 
-class DeterministicPickleCoderImpl(CoderImpl):
+class DeterministicCoderImpl(CoderImpl):
```

Comment:
I don't think a more general adapter is possible. It is unlikely we will have more than one implementation of a general purpose coder for most any serializable Python object, so having one would also not be useful past the already-supported use case enabled by PickleCoder / FastPrimitivesCoder.